src/CMakeLists.txt

```diff
@@ -357,31 +357,56 @@ target_compile_definitions(mudlet
     LUA_DEFAULT_PATH="${LUA_DEFAULT_DIR}")
 
 if(USE_OWN_QT5KEYCHAIN)
-    # An extra definition is needed if we are using our own QtKeyChain to not
-    # export symbols from our self-built library:
     target_compile_definitions(mudlet PRIVATE INCLUDE_OWN_QT5_KEYCHAIN QTKEYCHAIN_NO_EXPORT)
 endif()
 
-target_link_libraries(
-  mudlet
-  communi
-  edbee-lib
-  Boost::boost
-  HUNSPELL::HUNSPELL
-  LUA51::LUA51
-  PCRE::PCRE
-  PUGIXML::PUGIXML
-  Qt5::Concurrent
-  Qt5::Core
-  Qt5::Network
-  Qt5::Multimedia
-  Qt5::OpenGL
-  Qt5::UiTools
-  Qt5::Widgets
-  YAJL::YAJL
-  ZIP::ZIP
-  ZLIB::ZLIB
-  qt5keychain)
+if(WIN32 AND NOT USE_MAIN_BUILD_SYSTEM)
+  # For this combination the use of an INTERFACE definition (with the library
```

Comment:
This might sound a bit hard, but I have multiple issues with this:
1. Why the scope creep? Why not make MSYS2 working here with qmake (the build system we use on windows) Then add CMake changes in a follow-up PR?
2. Not using the interface definition seems to be working around insufficient find module definitions. Why don't we fix those instead of doing something half baked here?
3. If we must do the workaround, why not extract the differentiating libraries into a block, leaving the common code common?

Reply:
> 1. Why the scope creep? Why not make MSYS2 working here with qmake (the build system we use on windows) Then add CMake changes in a follow-up PR?

There is a separate thing that we want to also get into the project (a crash-handler) but that only comes with CMake build files at the moment - I am working on getting some QMake ones in upstream but I don't want to rely on that aspect going in straight away. Also, Windows is not my primary OS - and it is more convenient to me to work on both build systems whilst I am mentally in Windoze mode - and toggling between them helps to ensure that I am matching up the details/requirements for both equally - and allows me to compare the results so that they both work.

> 2. Not using the interface definition seems to be working around insufficient find module definitions. Why don't we fix those instead of doing something half baked here?

If you understand WTF is going on with those I would welcome your input - all I can say is that NOT using the `LIBRARYNAME::LIBRARYNAME` interface form for those libraries - which is what CMake seems to call it - fixes a bloody peculiar error - and that the CMake documentation is short on explanations or samples that actually show what is going on.

> 3. If we must do the workaround, why not extract the differentiating libraries into a block, leaving the common code common?

That is probably fair comment - I wasn't entirely sure that the order of listing was important initially - but now I realise it isn't so yes that can be made a smaller difference.

Reply:
2. :bulb: Actually, with regard to this point, since in the MSYS2/Mingw-w64 Windows platform most, if not all, the libraries involved ARE effectively system libraries as far as the compiler/linker toolchain is concerned (i.e. they come as part of the system) and in that situation we are not building the libraries from source code ourselves. So they must not have the interface type declaration (name repeated with a `::` between those) - instead they are IMPORTED libraries so are specified by their name in a singular form.

3. Done.

Reply:
> So they must not have the interface type declaration (name repeated with a :: between those) - instead they are IMPORTED libraries so are specified by their name in a singular form.

This is wrong. The `::` definitions are `IMPORTED` libraries as they should be. My guess is that the way you added the libraries, cmake is adding simply `-l hunspell` as linker flags. Which might work purely by chance, because the libraries are in the search path as expected. Transitive dependencies, linker flags or compile definitions are NOT honoured this way.

I may be able to help you get this right, but I'll need a clear way how to set up a MSYS2 compile environment for Mudlet.

Reply:
Ah, diving deeper down this rabbit hole it seems to me that both the IMPORTED and INTERFACE methods are for libraries that are built-with or at least accompany the main target in some way. However the libraries being separated out in this way are, when used in an MSYS2/Mingw-w64 Mudlet build, *normal* system-like libraries whereas if the original Windows build method is used they {the `hunspell`, `pcre`, `yajl` and `zip` libraries) are also built as part of the process.

Reply:
No. https://cmake.org/cmake/help/v3.18/command/add_library.html#imported-libraries explicitly states:
> An IMPORTED library target references a library file located outside the project. No rules are generated to build it

Reply:
🤷‍♂️ Okay, I am wrong in my theorising about what is happening. However whatever is the situation, the method I am using does *seem* to work. If it can be improved to be more like the other cases then I am not opposed to that - but so far using the LIBRARYNAME::LIBRARYNAME form for those libraries just breaks things in ways that the CMake documentation doesn't seem to cover...

Reply:
The underlying issue is now fixed with commit 6a95a41e1808ae51015986ade52a4bdaa7d8bad2.

When using the libraries in question, we had to differentiate between static and shared libraries. The static versions of these libraries required additional compile definitions, the shared versions did not.

Apparently, the detection, whether static or shared versions were used, was broken. This lead to the assumption of cmake, that the libraries were not identified correctly. This, in turn, made cmake add `LIBRARY::LIBRARY-NOTFOUND` to the Makefile, on which `make` choked.

Debugging was rather easy since the error message explicitly stated the line number on which `make` stopped. A few debug `MESSAGE` statements quickly identified the issue.